cnf-certification-test/chaostesting/suite.go

```diff
@@ -0,0 +1,308 @@
+		return "", err
+	}
+	output := bytes.ReplaceAll(input, []byte("{{APP_NAMESPACE}}"), []byte(namespace))
+	output = bytes.ReplaceAll(output, []byte("{{APP_LABEL}}"), []byte(appLabel))
+	output = bytes.ReplaceAll(output, []byte("{{APP_KIND}}"), []byte(appKind))
+	fileName := filename + ".tmp"
+	const filePermission = 0o600
+	if err = os.WriteFile(fileName, output, filePermission); err != nil {
+		logrus.Errorf("error: %s while writing the new template file %s ", err, fileName)
+		tnf.ClaimFilePrintf("error: %s while writing the new template file %s ", err, fileName)
+		return "", err
+	}
+	return fileName, nil
+}
+
+func waitForTestFinish(timeout time.Duration) bool {
+	const pollingPeriod = 1 * time.Second
+	var elapsed time.Duration
+	var result bool
+	for elapsed < timeout {
+		result = waitForResult()
+
+		if result {
+			break
+		}
+		time.Sleep(pollingPeriod)
+		elapsed += pollingPeriod
+	}
+	return result
+}
+
+func IsChaosResultVerdictPass() bool {
+	oc := clientsholder.GetClientsHolder()
+	gvr := schema.GroupVersionResource{Group: "litmuschaos.io", Version: "v1alpha1", Resource: "chaosresults"}
+	crs, err := oc.DynamicClient.Resource(gvr).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		logrus.Errorf("error getting : %v\n", err)
+	}
+	for _, cr := range crs.Items {
+		failResult := cr.Object["status"].(map[string]interface{})["experimentStatus"].(map[string]interface{})["failStep"]
+		verdictValue := cr.Object["status"].(map[string]interface{})["experimentStatus"].(map[string]interface{})["verdict"]
+		expKind := cr.Object["spec"].(map[string]interface{})["experiment"]
+		if expKind == chaosTestName {
+			if verdictValue == pass {
+				return true
+			}
+			logrus.Debugf("test completed but it failed with reason %s", failResult.(string))
+			return false
+		}
+	}
+	return false
+}
+
+func waitForResult() bool {
+	oc := clientsholder.GetClientsHolder()
+	gvr := schema.GroupVersionResource{Group: "litmuschaos.io", Version: "v1alpha1", Resource: "chaosengines"}
+	crs, err := oc.DynamicClient.Resource(gvr).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		logrus.Errorf("error getting : %v\n", err)
+	}
+	for _, cr := range crs.Items {
+		if status := cr.Object["status"]; status != nil {
+			if exp := status.(map[string]interface{})["experiments"]; exp != nil {
+				typ := exp.([]interface{})
+				status := cr.Object["status"].(map[string]interface{})["engineStatus"]
+				if typ[0].(map[string]interface{})["name"] == chaosTestName {
+					return status == completedResult
+				}
+			}
+		}
+	}
+	return false
+}
+
+//nolint:funlen //
+func createResource(filepath string) error {
+	oc := clientsholder.GetClientsHolder()
+	b, ferr := os.ReadFile(filepath)
+	if ferr != nil {
+		return ferr
+	}
+	k8sClient := oc.K8sClient
+	dynamicClient := oc.DynamicClient
+	const oneh = 100
+	decoder := yamlutil.NewYAMLOrJSONDecoder(bytes.NewReader(b), oneh)
+	for {
```

Comment:
since we have more than one kind on the service-account file so we need to go one by one and maybe in other files it will have a number of kind to create that we dont know how much